Core.ApplicationServices/SingleThreadedMailClient.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Core.DomainServices;
+using System.Net.Mail;
+using Infrastructure.Services.Delegates;
+
+namespace Core.ApplicationServices
+{
+    public class SingleThreadedMailClient : IMailClient
+    {
+        /// <summary>
+        /// Overcome restrictions defined by the target service e.g. services such as Office365: https://techcommunity.microsoft.com/t5/exchange-team-blog/changes-coming-to-the-smtp-authenticated-submission-client/ba-p/607825
+        /// </summary>
+        private static readonly object ConnectionLock = new object();
+
+        private readonly Factory<SmtpClient> _clientFactory;
+
+        /// <summary>
+        /// Construct a smtp client with configuration from web.config
+        /// </summary>
+        public SingleThreadedMailClient()
+        {
+            _clientFactory = () => new SmtpClient();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SingleThreadedMailClient"/> class.
+        /// </summary>
+        /// <param name="host">The host.</param>
+        /// <param name="port">The port.</param>
+        /// <param name="enableSsl">if set to <c>true</c> [enable SSL].</param>
+        public SingleThreadedMailClient(string host, int port, bool enableSsl = false)
+        {
+            _clientFactory = () => new SmtpClient(host, port) { EnableSsl = enableSsl };
+        }
+
+        /// <summary>
+        /// Sends the specified mail.
+        /// </summary>
+        /// <param name="message">The message.</param>
+        public virtual void Send(MailMessage message)
+        {
+            lock (ConnectionLock)
```

Comment:
Det er her nede forskellene ses. For det første er mail client ikke længere disposable for nu håndteres lifetime af smtpclient direkte i metoden her. For det andet er adgangen til at kunne sende mails nu synkroniseret uanset mængden af instanser af klassen (ConnectionLock) Dette modvirker det connection constraint problem der har ført til alle fejlene siden vi skiftede fra almindelig smtp til Office365